statsmodels/robust/tools.py

```diff
@@ -0,0 +1,234 @@
+    eff : float in (0, 1)
+        Required asymptotic relative efficiency compared to least squares
+        at the normal reference distribution. For example, ``eff=0.95`` for
+        95% efficiency.
+    kwd : str
+        Name of keyword for tuning parameter.
+    kwargs : dict or None
+        Dict for other keyword parameters.
+    use_jump : bool
+        If False (default), then use computation that require continuous
+        psi function.
+        If True, then use computation then the psi function can have jump
+        discontinuities.
+    bracket : None or tuple
+        Bracket with lower and upper bounds to use for scipy.optimize.brentq.
+        If None, than a default bracket, currently [0.1, 10], is used.
+
+    Returns
+    -------
+    Float : Value of tuning parameter to achieve asymptotic relative
+        efficiency.
+
+    """
+    kwds = {} if kwargs is None else kwargs
+    if bracket is None:
+        bracket = [0.1, 10]
+
+    if not use_jump:
+        def func(c):
+            kwds.update({kwd: c})
+            return _var_normal(norm(**kwds) - 1 / eff)
+    else:
+        def func(c):
+            kwds.update({kwd: c})
+            return _var_normal_jump(norm(**kwds) - 1 / eff)
+
+    res = optimize.brentq(func, *bracket)
+    return res[0]
+
+
+def tuning_s_estimator_mean(norm, breakdown=None):
+    """Tuning parameter and scale bias correction for S-estimators of mean.
+
+    The reference distribution is the normal distribution.
+
+    Parameters
+    ----------
+    norm : RobustNorm subclass
+    breakdown : float or iterable of float in (0, 0.5]
+        Desired breakdown point between 0 and 0.5.
+        Default if breakdown is None is a list of breakdown points.
+
+    Returns
+    -------
+    Holder instance with the following attributes :
+
+     - `breakdown` : breakdown point
+     - `eff` : relative efficiency
+     - `param` : tuning parameter for norm
+     - `scale_bias` : correction term for Fisher consistency.
+
+    Notes
+    -----
+    Based on Rousseeuw and Leroy (1987). See table 19, p. 142 that can be
+    replicated by this function for TukeyBiweight norm.
+    Note, the results of this function are based computation without rounding
+    to decimal precision, and differ in some cases in the last digit from
+    the table by Rousseeuw and Leroy.
+
+    Numerical expectation and root finding based on scipy integrate and
+    optimize.
+
+    TODO: more options for details, numeric approximation and root finding.
+
+    Reference
+    ---------
+    Rousseeuw and Leroy book
+
+
+    """
+    if breakdown is None:
+        bps = [0.5, 0.45, 0.40, 0.35, 0.30, 0.25, 0.20, 0.15, 0.1, 0.05]
+    else:
+        # allow for scalar bp
+        try:
+            _ = iter(breakdown)
+            bps = breakdown
+        except TypeError:
+            bps = list(breakdown)
+
+    def func(c):
+        norm = norm(c=c)
+        bp = stats.norm.expect(lambda x : norm.rho(x)) / norm.rho(norm.c)
```

Comment:
this will not work across norms

norm.c is not always the argmax rho, and in several cases the keyword/attribute name differs